fission/src/ui/panels/configuring/scoring/ZoneConfigPanel.tsx

```diff
@@ -1,84 +1,202 @@
-    // somehow get and store which zone is being edited
-    // maybe a global ConfigProvider in App.tsx?
-    // then set all default values to the state of the zone
-    const [, setName] = useState<string>("")
-    const [alliance, setAlliance] = useState<"red" | "blue">("blue")
-    const [, setParent] = useState<string>("")
-    const [, setPoints] = useState<number>(1)
-    const [, setDestroy] = useState<boolean>(false)
-    const [, setPersistent] = useState<boolean>(false)
-    const [, setScale] = useState<[number, number, number]>([1, 1, 1])
+    const { openPanel, closePanel } = usePanelControlContext()
+
+    const transformGizmoRef = useRef<TransformGizmos>()
+
+    const [name, setName] = useState<string>(SelectedZone.zone.name)
+    const [alliance, setAlliance] = useState<"red" | "blue">(SelectedZone.zone.alliance)
+    const [selectedNode, setSelectedNode] = useState<string | undefined>(SelectedZone.zone.parentNode)
+    const [points, setPoints] = useState<number>(SelectedZone.zone.points)
+    const [destroy, setDestroy] = useState<boolean>(SelectedZone.zone.destroyGamepiece)
+    const [persistent, setPersistent] = useState<boolean>(SelectedZone.zone.persistentPoints)
+
+    const [transformMode, setTransformMode] = useState<"translate" | "rotate" | "scale">("translate")
+
+    const { currentTheme, themes } = useTheme()
+    const theme = useMemo(() => {
+        return themes[currentTheme]
+    }, [currentTheme, themes])
+
+    const field = useMemo(() => {
+        const assemblies = [...World.SceneRenderer.sceneObjects.values()]
+        for (let i = 0; i < assemblies.length; i++) {
+            const assembly = assemblies[i]
+            if (!(assembly instanceof MirabufSceneObject)) continue
+
+            if ((assembly as MirabufSceneObject).miraType != MiraType.FIELD) continue
+
+            return assembly
+        }
+
+        return undefined
+    }, [])
+
+    useEffect(() => {
+        closePanel("scoring-zones")
+        configureGizmo()
+    }, [])
+
+    const configureGizmo = () => {
+        transformGizmoRef.current = new TransformGizmos(
+            new THREE.Mesh(
+                new THREE.BoxGeometry(1, 1, 1),
+                World.SceneRenderer.CreateToonMaterial(ReactRgbaColor_ThreeColor(theme.HighlightHover.color))
+            )
+        )
+        transformGizmoRef.current.AddMeshToScene()
+
+        transformGizmoRef.current.CreateGizmo("translate", 1.5)
+
+        const position = SelectedZone.zone.position
+        const rotation = SelectedZone.zone.rotation
+        const scale = SelectedZone.zone.scale
+
+        transformGizmoRef.current.mesh.position.set(position[0], position[1], position[2])
+        transformGizmoRef.current.mesh.rotation.setFromQuaternion(
+            new THREE.Quaternion(rotation[0], rotation[1], rotation[2], rotation[3])
+        )
+        transformGizmoRef.current.mesh.scale.set(scale[0], scale[1], scale[2])
+    }
```

Comment:
Just make this an effect instead of create a function that is ran in an effect.

Reply:
Every time this panel rerenders, it is recreating this function, despite only calling it once.